LineStream/Stream.ts

```diff
@@ -0,0 +1,86 @@
+import { CommandBuilder } from "../deps.ts";
+import { LineStream, XargsFunction } from "./LineStream.ts";
+import { ApplyFunction, FilterFunction, MapFunction } from "./Transformer.ts";
+import { XargsStream } from "./XargsStream.ts";
+
+export interface StreamInterface {
+  /**
+   * Async iterator implementation for iterating over the lines of the stream.
+   * @returns An async iterator for iterating over the lines of the stream.
+   */
+  [Symbol.asyncIterator](): AsyncGenerator;
+
+  /**
+   * Reads the entire stream and returns the concatenated text.
+   * @returns A promise that resolves to the concatenated text.
+   */
+  text(): Promise<string>;
+
+  /**
+   * Reads the entire stream and returns an array of lines.
+   * @returns A promise that resolves to an array of lines.
+   */
+  lines(): Promise<string[]>;
+
+  // TODO: should this be added ?
```

Comment:
not sure about this part, probably should be implementd, but how should it work in xargs? just `this.linestream.pipethrough` ?

Reply:
I think this.linestream.pipethrough is fine:+1;